### PR TITLE
Feature/add association

### PR DESCRIPTION
**Pull request by PabloBona**
**Changes Made:**

1. Implemented the `Classroom` class with the following features:
   - Added an `@label` instance variable, initialized in the constructor.
   - Created setter and getter methods for `@label` using `attr_accessor`.
   - Established the has-many relationship with `Student` class.
   - Ensured that adding a student to a classroom also sets the classroom for the student.

2. Developed the `Student` class with the following features:
   - Created an `@classroom` attribute with getter and setter methods.
   - Ensured that setting the classroom for a student adds the student to the classroom's students.

3. Created the `Book` class with the following attributes and methods:
   - `@title` and `@author` instance variables initialized in the constructor.
   - Setter and getter methods for the instance variables using `attr_accessor`.
   - Defined a `@rentals` array to track associated rentals.
   - Implemented the `add_rental` method to add a rental to the book's rentals.

4. Constructed the `Rental` class with the following attributes and methods:
   - `@date` instance variable initialized in the constructor.
   - Setter and getter methods for the `@date` attribute.
   - Established the many-to-many relationship between `Person` and `Book` using `Rental` as an intermediate class.
   - Implemented the required has-many and belongs-to relationships for `Person`, `Book`, and `Rental`.
   - Modified the constructor to set both `Book` and `Person` when creating a new `Rental`.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
-require_relative 'person'
-
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom&.add_student(self)
   end
```
Comment (Owusu-Desmond) [inline student.rb]: - [ ] I suggest that you add a method that serves as the 'belongs-to' side (a student belongs to a classroom). Link to the article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`
  
  ```ruby
  def classroom=(classroom)
        @classroom = classroom
        classroom.students.push(self) unless classroom.students.include?(self)
    end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (Owusu-Desmond) [inline book.rb]: - [ ] I think this method for adding rentals should take in the `person` and the `date` and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.` like this :point_down:
  
  ```ruby
  # *A method for adding rentals should be present.
  def add_rental(person, date)
      # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
      Rental.new(date, self, person)
    end
  ```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+    @person = person
+    @book = book
+
+    person.add_rental(self)
```
Comment (Owusu-Desmond) [inline rental.rb]: - [ ] Please ensure that the rental is adding itself to the @book achieving the 'belongs-to' association. :pray:
  ```ruby
      book.rentals << self
  ```

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    book.add_rental(self)
```
Comment (Owusu-Desmond) [inline rental.rb]: - [ ] Please ensure that the rental is adding itself to the @person the 'belongs-to' association. :pray:
  ```ruby
      person.rentals << self
  ```

File: person.rb [hunk 5]
```diff
@@ -7,15 +7,20 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
   end
 
-  attr_reader :id, :age
+  attr_reader :id, :age, :rentals
   attr_accessor :name
 
+  def add_rental(rental)
+    @rentals << rental
+  end
+
```
Comment (Owusu-Desmond) [inline person.rb]: - [ ] I suggest that this method for adding rentals takes in the `book` and the `date` and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    
    ```ruby
      def add_rental(book, date)
        Rental.new(date, book, self)
      end
    ```

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @PabloBona , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 4 above)
Comment (PabloBona) [reply rental.rb]: Wow, thanks for the feedback and for taking the time to explain me step by step! I really appreciated that.
Changes are done!

File: person.rb (hunk 5 above)
Comment (PabloBona) [reply person.rb]: Wow, thanks for the feedback and for taking the time to explain me step by step! I really appreciated that.
Changes are done!

**Review by OLIPLICHE — APPROVED** [review]
Hi @PabloBona, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Apply the changes requested from the previous review (✅)) 👏
- Set up associations between classes👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**